### DISCUSSION TITLE
Registering a syntax leads to "A 'XYZ' is not a registered object."

### DISCUSSION THREAD

**Discussion by jmeier** [idaholab/moose#27186] [Q&A General] [answered]
Dear all,

In "registerAll" of my moose app I try to register an input file syntax like `[XYZ]` (on top level) using `registerSyntaxTask("AddXYZAction", "XYZ", "add_xyz");`.

The app compiles just fine.

My input file:
```
[XYZ]
  type = 'XYZ'
[]
```

If I run my app using the input file above I get
```
*** ERROR ***
A 'XYZ' is not a registered object.
```

Do you know why my Moose app is not finding the syntax I registered?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Did you register a XYZ object to your application?
The syntax is recognized but you also need an object with that type

Guillaume

**Reply by jmeier** [reply]
Dear @GiudGiud ,

Thanks - you are right. I forgot to add:
`registerMooseObject("moose_bare_tmechApp", XYZ);`
